shardingsphere-infra/shardingsphere-infra-federation/shardingsphere-infra-federation-executor/src/main/java/org/apache/shardingsphere/infra/federation/executor/advanced/table/TranslatableTableScanExecutor.java

```diff
@@ -0,0 +1,279 @@
+            for (JDBCExecutionUnit executionUnit : each.getInputs()) {
+                result.add(executionUnit.getStorageResource());
+            }
+        }
+        return result;
+    }
+    
+    private SqlString createSQLString(final ShardingSphereTable table, final TranslatableScanNodeExecutorContext scanContext, final SqlDialect sqlDialect) {
+        return new RelToSqlConverter(sqlDialect).visitRoot(createRelNode(table, scanContext)).asStatement().toSqlString(sqlDialect);
+    }
+    
+    private void setParameters(final Collection<ExecutionGroup<JDBCExecutionUnit>> inputGroups) {
+        for (ExecutionGroup<JDBCExecutionUnit> each : inputGroups) {
+            for (JDBCExecutionUnit executionUnit : each.getInputs()) {
+                if (!(executionUnit.getStorageResource() instanceof PreparedStatement)) {
+                    continue;
+                }
+                setParameters((PreparedStatement) executionUnit.getStorageResource(), executionUnit.getExecutionUnit().getSqlUnit().getParameters());
+            }
+        }
+    }
+    
+    @SneakyThrows(SQLException.class)
+    private void setParameters(final PreparedStatement preparedStatement, final List<Object> parameters) {
+        for (int i = 0; i < parameters.size(); i++) {
+            Object parameter = parameters.get(i);
+            preparedStatement.setObject(i + 1, parameter);
+        }
+    }
+    
+    private RelNode createRelNode(final ShardingSphereTable table, final TranslatableScanNodeExecutorContext scanContext) {
+        String databaseName = executorContext.getDatabaseName();
+        String schemaName = executorContext.getSchemaName();
+        CalciteConnectionConfig connectionConfig = new CalciteConnectionConfigImpl(OptimizerPlannerContextFactory.createConnectionProperties());
+        ShardingSphereSchema schema = executorContext.getFederationContext().getDatabases().get(databaseName).getSchema(schemaName);
+        CalciteCatalogReader catalogReader = OptimizerPlannerContextFactory.createCatalogReader(schemaName,
+                new FilterableSchema(schemaName, schema, null), new JavaTypeFactoryImpl(), connectionConfig);
+        RelOptCluster relOptCluster = RelOptCluster.create(QueryOptimizePlannerFactory.createVolcanoPlanner(), new RexBuilder(new JavaTypeFactoryImpl()));
+        RelBuilder builder = RelFactories.LOGICAL_BUILDER.create(relOptCluster, catalogReader).scan(table.getName());
+        if (null != scanContext.getFilterValues()) {
+            builder.filter(createFilters(scanContext.getFilterValues(), builder, table.getColumnNames()));
+        }
+        if (null != scanContext.getProjects()) {
+            builder.project(createProjections(scanContext.getProjects(), builder, table.getColumnNames()));
+        }
+        return builder.build();
+    }
+    
+    private Collection<RexNode> createFilters(final String[] filterValues, final RelBuilder builder, final List<String> columnNames) {
```

Comment:
Can we move this logic to the entrance, and remove TranslatableTableScanExecutor?